flink-core/src/main/java/org/apache/flink/configuration/TaskManagerOptions.java

```diff
@@ -217,9 +217,22 @@
 				.build());
 
 	// ------------------------------------------------------------------------
-	//  Memory Options
+	//  Resource Options
 	// ------------------------------------------------------------------------
 
+	/**
+	 * This config key is used for passing cpu cores into TaskExecutors.
+	 * There're YarnConfigOptions#VCORES and MesosTaskManagerParameters#MESOS_RM_TASKS_CPUS for configuring TM container
+	 * CPU on Yarn and Mesos respectively.
+	 */
+	public static final ConfigOption<Double> CPU_CORES =
+		key("taskmanager.cpu.cores")
+			.doubleType()
+			.defaultValue(-1.0)
```

Comment:
I'm in favor of make it no default.
Setting it to 1 by default may break the behavior on standalone clusters. Although we don't have any task that specify their cpu requirements at the moment, once we have it may prevent us from deploying more tasks onto the task executor if the machine does have more than one core but the user did not explicitly configure it.

Reply:
As discussed offline, I'll make the this config option no default, and fall back to number of slots (1 / default-slot-fraction) if not configured.